curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -73,8 +73,10 @@
     private final AtomicReference<byte[]> data = new AtomicReference<byte[]>();
     private final AtomicReference<State> state = new AtomicReference<State>(State.LATENT);
     private final AtomicBoolean authFailure = new AtomicBoolean(false);
+    private volatile boolean parentCreationFailure;
```

Comment:
Should this be an AtomicBoolean for consistency with the approach used for authFailure?

Reply:
This is what I thought, but making it an `AtomicBoolean` seems overkill since we're not doing any compare and set operations. Maybe we could just make the `authFailure` field volatile?

Reply:
Yeah, I think that's a reasonable alternative.

Reply:
Ok I updated it.